### PR TITLE
Add --ignore-warnings to Export command

### PR DESCRIPTION
**Pull request by Trenly**
- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] This pull request is related to an issue.
  * #4213

Manually validated output files remained the same.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4214)

### CODE REVIEW

File: src/AppInstallerCLICore/Workflows/ImportExportFlow.cpp [hunk 1]
```diff
@@ -98,6 +98,7 @@ namespace AppInstaller::CLI::Workflow
     {
         const auto& searchResult = context.Get<Execution::Data::SearchResult>();
         const bool includeVersions = context.Args.Contains(Execution::Args::Type::IncludeVersions);
+        const bool suppressWarnings = context.Args.Contains(Execution::Args::Type::IgnoreWarnings);
```
Comment (yao-msft) [inline src/AppInstallerCLICore/Workflows/ImportExportFlow.cpp:101]: Should this be applied to line 89 as well?

Comment (Trenly) [reply src/AppInstallerCLICore/Workflows/ImportExportFlow.cpp:104]: Makes sense to me

**Review by yao-msft — APPROVED** [review]

**Comment by yao-msft** [comment]
Just leave my thoughts here. In the future, if we want to filter the output, we should do a refactor and do it in reporter level. So we don't need to check every place. Something like Reporter.DisableReportLevel(Level::Warning), then reporter will no longer output for warning level.

**Comment by Trenly** [comment]
> Just leave my thoughts here. In the future, if we want to filter the output, we should do a refactor and do it in reporter level. So we don't need to check every place. Something like Reporter.DisableReportLevel(Level::Warning), then reporter will no longer output for warning level.

Good point; I can try taking a look into that (in a separate PR), though it may be above my skill level

**Comment by yao-msft** [comment]
> Good point; I can try taking a look into that (in a separate PR), though it may be above my skill level

If we can do in the Reporter level, I would discard this pr. This one only disable the warning in this specific workflow task, it will not ignore warnings from previous steps in ExportCommand like HandleSearchResultFailures 

And we don't need to do spot checks at multiple places. the code will be cleaner.

**Comment by Trenly** [comment]
If we can do it at the reporter level, I would revert these changes as part of the PR to handle the reporter refactor, since with that refactor the `--ignore-warnings` arg should likely be added to the common args list rather than individual commands anyways

**Comment by yao-msft** [comment]
> If we can do it at the reporter level, I would revert these changes as part of the PR to handle the reporter refactor, since with that refactor the `--ignore-warnings` arg should likely be added to the common args list rather than individual commands anyways

That's fine too.

**Comment by Trenly** [comment]
```
Failed ConfigureFromHttpsConfigurationFile [282 ms]
  Error Message:
     Expected: 0
  But was:  1
  Stack Trace:
     at AppInstallerCLIE2ETests.ConfigureCommand.ConfigureFromHttpsConfigurationFile() in D:\a\1\s\src\AppInstallerCLIE2ETests\ConfigureCommand.cs:line 182
```

**Pull request closed without merging**